### ISSUE TITLE
Syntax error when transpiling JSX in version 1.6.12

### ISSUE THREAD

**Issue by benStre** [swc-project/swc#9172]
### Describe the bug

Since the 1.6.12 release, SWC shows an error message when using JSX syntax. The code is still correctly transpiled, but the error messages are a bit confusing.

Error message:
> ERROR  Syntax error called from /Users/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/swc_ecma_parser-0.146.11/src/parser/typescript.rs:151:13
> 
> Current token = Some(/)

### Input code

```tsx
console.log(<div/>)
```

### Config

```json
{
    "jsc": {
      "parser": {
        "tsx": true,
        "syntax": "typescript"
      }
    }
}
```

### Playground link (or link to the minimal reproduction)

https://github.com/unyt-org/swc-jsx-error-example

### Expected behavior

The code should be transpiled without showing an error message.

### Actual behavior

An error message is displayed:
> ERROR  Syntax error called from /Users/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/swc_ecma_parser-0.146.11/src/parser/typescript.rs:151:13
>
> Current token = Some(/)

### Version

1.6.12

### Additional context

I am using the SWC npm module in deno (`import {transform} from "npm:@swc/core@1.6.12"`)

**Comment by magic-akari** [comment]
I suspect it was introduced by #9141 and fixed by #9171. Let's wait for a nightly build to verify this issue.

**Comment by fvilers** [comment]
Same issue using vite 5.3.3 with Node 18.x on Windows 10.

```
  ERROR  Syntax error called from C:\Users\runneradmin/.cargo\registry\src\index.crates.io-6f17d22bba15001f\swc_ecma_parser-0.146.11\src\parser\typescript.rs:151:13
Current token = Some(.)
    at C:\Users\runneradmin/.cargo\registry\src\index.crates.io-6f17d22bba15001f\swc_ecma_parser-0.146.11\src\parser\typescript.rs:151

  ERROR  Syntax error called from C:\Users\runneradmin/.cargo\registry\src\index.crates.io-6f17d22bba15001f\swc_ecma_parser-0.146.11\src\parser\expr.rs:116:26
Current token = Some(Lists)
    at C:\Users\runneradmin/.cargo\registry\src\index.crates.io-6f17d22bba15001f\swc_ecma_parser-0.146.11\src\parser\expr.rs:116

  ERROR  Syntax error called from C:\Users\runneradmin/.cargo\registry\src\index.crates.io-6f17d22bba15001f\swc_ecma_parser-0.146.11\src\parser\expr.rs:493:9 
Current token = Some(<lexing error: Error { error: (363..368#0, UnterminatedRegExp) }>)
    at C:\Users\runneradmin/.cargo\registry\src\index.crates.io-6f17d22bba15001f\swc_ecma_parser-0.146.11\src\parser\expr.rs:49
```

**Comment by magic-akari** [comment]
Fixed in [1.6.13](https://www.npmjs.com/package/@swc/core/v/1.6.13).
